packages/netlify-cms-core/src/reducers/editorialWorkflow.ts

```diff
@@ -66,48 +69,64 @@ const unpublishedEntries = (state = Map(), action: EditorialWorkflowAction) => {
       });
 
     case UNPUBLISHED_ENTRY_PERSIST_REQUEST: {
+      // Update Optimistically
+      return state.setIn(
+        [
+          'entities',
+          `${action.payload!.collection}.${action.payload!.entry.get('slug')}`,
+          'isPersisting',
+        ],
+        true,
+      );
+    }
+
+    case UNPUBLISHED_ENTRY_PERSIST_SUCCESS:
       // Update Optimistically
       return state.withMutations(map => {
         map.setIn(
           ['entities', `${action.payload!.collection}.${action.payload!.entry.get('slug')}`],
           fromJS(action.payload!.entry),
         );
-        map.setIn(
-          [
-            'entities',
-            `${action.payload!.collection}.${action.payload!.entry.get('slug')}`,
-            'isPersisting',
-          ],
-          true,
-        );
+        map.deleteIn([
+          'entities',
+          `${action.payload!.collection}.${action.payload!.slug}`,
+          'isPersisting',
+        ]);
         map.updateIn(['pages', 'ids'], List(), list =>
           list.push(action.payload!.entry.get('slug')),
         );
       });
-    }
 
-    case UNPUBLISHED_ENTRY_PERSIST_SUCCESS:
-      // Update Optimistically
-      return state.deleteIn([
-        'entities',
-        `${action.payload!.collection}.${action.payload!.slug}`,
-        'isPersisting',
-      ]);
+    case UNPUBLISHED_ENTRY_PERSIST_FAILURE:
+      return state.setIn(
+        [
+          'entities',
+          `${action.payload!.collection}.${action.payload!.entry.get('slug')}`,
```

Comment:
I think `entry` is missing from the payload:
https://github.com/netlify/netlify-cms/pull/4289/files#diff-f54d023791c4074f77a150bb2ec73579R147